test/test.ts

```diff
@@ -35,6 +35,15 @@ describe('devalue', () => {
 		test('Map', new Map([['a', 'b']]), 'new Map([["a","b"]])');
 	});
 
+	describe('strings', () => {
+		test('newline', 'a\nb', JSON.stringify('a\nb'));
+		test('double quotes', '"yar"', JSON.stringify('"yar"'));
+		test('lone surrogate', "\uD800", '"\\uD800"');
```

Comment:
Tests I'd consider adding for this change (with the ones that are already there marked as completed):

- [x] a surrogate pair, e.g. `'a\uD800\uDC00b'`
- [ ] a surrogate pair in the wrong order, e.g. `'a\uDC00\uD800b'`
- [x] a lone high surrogate, e.g. `'a\uD800b'`
- [ ] a lone low surrogate, e.g. `'a\uDC00b'`
- [ ] two lone high surrogates in a row, e.g. `'a\uD800\uD800b'`
- [ ] two lone low surrogates in a row, e.g. `'a\uDC00\uDC00b'`

Reply:
I've added these tests, though I'm not entirely clear on what the expected values are 😂 — should any combination of surrogates other than [high, low] be replaced with `\\u[XXXX]`?

Reply:
Exactly; anything that is not [high, low] is not a valid surrogate pair, but rather a series of lone surrogates.

When in doubt, you can compare the results to `JSON.stringify` in a recent V8 build (perhaps using [jsvu](https://github.com/GoogleChromeLabs/jsvu) to easily grab the latest binary).